cmd/fetchd/cmd/capricornmigrate.go

```diff
@@ -0,0 +1,489 @@
+			if err != nil {
+				return fmt.Errorf("failed to burn tokens: %w", err)
+			}
+
+			// Enable IBC
+			numHistoricalEntries, err := cmd.Flags().GetUint32(flagStakingParamsHistoricalEntries)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve flag %q: %w", flagStakingParamsHistoricalEntries, err)
+			}
+
+			appState, err = enableIBC(appState, cdc, numHistoricalEntries)
+			if err != nil {
+				return fmt.Errorf("failed to enable IBC: %w", err)
+			}
+
+			// removes unused bridge contract
+			appState, err = deleteWrongBridgeContract(appState, cdc)
+			if err != nil {
+				return fmt.Errorf("failed to delete wrong bridge contract: %w", err)
+			}
+
+			// update bridge and mobix contract codes
+			newBridgeContractPath, err := cmd.Flags().GetString(flagBridgeNewContractPath)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve flag %q: %w", flagBridgeNewContractPath, err)
+			}
+			newMobixContractPath, err := cmd.Flags().GetString(flagMobixNewContractPath)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve flag %q: %w", flagMobixNewContractPath, err)
+			}
+
+			appState, err = migrateWasmContracts(appState, cdc, newBridgeContractPath, newMobixContractPath)
+			if err != nil {
+				return fmt.Errorf("failed to migrate wasm contracts: %w", err)
+			}
+
+			appState, err = updateMobixContractState(appState, cdc)
+			if err != nil {
+				return fmt.Errorf("failed to migrate wasm contracts: %w", err)
+			}
+
+			slashingParamDoubleSignFractionStr, err := cmd.Flags().GetString(flagNewSlashFractionDoubleSign)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve flag %q: %w", flagNewSlashFractionDoubleSign, err)
+			}
+
+			slashingParamDoubleSignFraction, err := sdk.NewDecFromStr(slashingParamDoubleSignFractionStr)
+			if err != nil {
+				return fmt.Errorf("invalid flag %q value: %w", flagNewSlashFractionDoubleSign, err)
+			}
+
+			appState, err = setNewSlashFractionDoubleSign(appState, cdc, slashingParamDoubleSignFraction)
+			if err != nil {
+				return fmt.Errorf("failed to set new slashing double sign fraction param: %w", err)
+			}
+
+			// Validate state (same as fetchd validate-genesis cmd)
+			if err := app.ModuleBasics.ValidateGenesis(cdc, clientCtx.TxConfig, appState); err != nil {
+				return fmt.Errorf("failed to validate state: %w", err)
+			}
+
+			// build and print the new genesis state
+			genDoc.AppState, err = json.Marshal(appState)
+			if err != nil {
+				return errors.Wrap(err, "failed to JSON marshal migrated genesis state")
+			}
+			bz, err := tmjson.Marshal(genDoc)
+			if err != nil {
+				return errors.Wrap(err, "failed to marshal genesis doc")
+			}
+			sortedBz, err := sdk.SortJSON(bz)
+			if err != nil {
+				return errors.Wrap(err, "failed to sort JSON genesis doc")
+			}
+
+			fmt.Println(string(sortedBz))
+
+			return nil
+		},
+	}
+
+	cmd.Flags().String(flagGenesisTime, "", "override genesis_time with this flag")
```

Comment:
Is there a specific reason not to use `cmd.Flags().StringVar(&flagGenesisTime`, ...)` (`Int64Var`, etc.)? I think we then wouldn't need to do `cmd.Flags().GetString(...)` (etc.) as the "flag var" will hold the value.

Reply:
This should probably work, I think I mostly avoid it because of the different context we're defining the flag and retrieving their value (switching from https://github.com/fetchai/fetchd/blob/capricorn_migration/cmd/fetchd/cmd/capricornmigrate.go#L77). Having them like this would ease moving this RunE callback to another file or struct without having to change too much stuff.

Reply:
That's a good point, I hadn't considered :smile: